### PR TITLE
Chore/fix eslint issues/152798539

### PR DESCRIPTION
**Pull request by ipheghe**
### What does this PR do?
Fix eslint errors on client folder

### Description of Task to be completed?
- fix eslint issues for files in client folder

### How should this be manually tested?
N/A

### Any background context you want to provide?
N/A

## What are the relevant pivotal tracker stories?
story type: chore
story Id: 152798539
story title: Fix all eslint problems for  files in client folder

### Screenshots (if appropriate)
N/A

### Questions:
N/A

### CODE REVIEW

**Comment by andela-cofor** [comment]
LGTM.

File: client/src/actions/auth.js [hunk 1]
```diff
@@ -9,13 +15,7 @@ import {
...
-import {
-  actions as toastrActions
-} from 'react-redux-toastr';
-import {
-  bindActionCreators
-} from 'redux';
-import jwtDecode from 'jwt-decode';
+import decodeToken from '../../../server/helpers/decodeToken';
 
 //= ===============================
```
Comment (nelsonabienorin) [inline client/src/actions/auth.js:20]: You can remove unnecessary comments

File: client/src/actions/auth.js [hunk 2]
```diff
@@ -100,7 +96,7 @@ export function registerUser({
           dispatch({
             type: AUTH_USER
           });
-          location.hash = '#login';
+          window.location.hash = '#login';
```
Comment (nelsonabienorin) [inline client/src/actions/auth.js:99]: why are you using window.location.hash?

File: client/src/actions/favorite.js [hunk 3]
```diff
@@ -27,15 +27,17 @@ const favoriteRecipe = (recipeId, categoryId) => {
   const directTo = '';
   const toastMessage = 'Recipe added to favorites Successfully';
   const constant = 'RECIPE_FAVORITED';
```
Comment (nelsonabienorin) [inline client/src/actions/favorite.js:29]: names that are similar to reserved words should be avoided as variable name.

File: client/src/actions/index.js [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
-export const API_URL = 'http://localhost:8000/api/v1';
-export const CLIENT_ROOT_URL = 'http://localhost:3000';
 import {
   actions as toastrActions
 } from 'react-redux-toastr';
 import {
   bindActionCreators
 } from 'redux';
 
+export const API_URL = 'http://localhost:8000/api/v1';
```
Comment (nelsonabienorin) [inline client/src/actions/index.js:9]: Why are you setting localhost as your URL here?

File: client/src/actions/index.js [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
+export const CLIENT_ROOT_URL = 'http://localhost:3000';
+
 //= ===============================
 // Utility actions
```
Comment (nelsonabienorin) [inline client/src/actions/index.js:13]: Your comment could be formatted in a better way with jsdoc or removed if not necessary

File: client/src/actions/index.js [hunk 6]
```diff
@@ -71,7 +71,7 @@ export const postData = (
   if (isAuthReq) {
     headers = {
       headers: {
-        'x-access-token': localStorage.getItem('token')
+        'x-access-token': window.localStorage.getItem('token')
```
Comment (nelsonabienorin) [inline client/src/actions/index.js:74]: Is there any difference between `window.localStorage.getItem('token')` and  `localStorage.getItem('token')`?

File: client/src/actions/index.js [hunk 7]
```diff
@@ -20,7 +21,6 @@ import {
  * @returns {*} void
  */
 export function errorHandler(dispatch, error, type) {
```
Comment (nelsonabienorin) [inline client/src/actions/index.js:23]: ES6 function could be used

File: client/src/actions/index.js [hunk 8]
```diff
@@ -189,7 +189,7 @@ export const putData = (
         payload: response.data,
       });
       if (directTo.length > 3) {
```
Comment (nelsonabienorin) [inline client/src/actions/index.js:191]: why are you checking if length of `directTo` is > 3?

File: client/src/localStore.js [hunk 9]
```diff
@@ -1,6 +1,8 @@
-export const get = () => JSON.parse(localStorage.getItem('state')) || undefined;
+export const get = () => JSON.parse(window.localStorage.getItem('state')) || undefined;
 export const set = (state, props) => {
-  let toSave = {};
-  props.forEach(p => toSave[p] = state[p]);
-  localStorage.setItem('state', JSON.stringify(toSave));
-};
\ No newline at end of file
+  const toSave = {};
+  props.forEach((p) => {
```
Comment (nelsonabienorin) [inline client/src/localStore.js:4]: whats is p?

**Review by nelsonabienorin — COMMENTED** [review]
Good job so far, kindly address the comments and make changes where necessary

File: client/src/actions/index.js (hunk 7 above)
Comment (ipheghe) [reply client/src/actions/index.js:23]: ok. will be addressed

File: client/src/actions/index.js (hunk 6 above)
Comment (ipheghe) [reply client/src/actions/index.js:74]: made window as a global in my eslint config... local storage used like that will work but my eslint will flag it that it is not defined

File: client/src/actions/index.js (hunk 8 above)
Comment (ipheghe) [reply client/src/actions/index.js:191]: directTo is the next location after an action has been performed, so am checking if the directTo value length is more than 3. if so it directs user to the next page while if it is not it does not direct user to another page

File: client/src/actions/auth.js (hunk 2 above)
Comment (ipheghe) [reply client/src/actions/auth.js:99]: am using hash router..can only access my routes location through location.hash

File: client/src/actions/favorite.js (hunk 3 above)
Comment (ipheghe) [reply client/src/actions/favorite.js:29]: can i use toastrConstant for the variable name

**Pull request merged**